sdk/python/test_picovoice.py

```diff
@@ -21,89 +18,101 @@
+                access_key=sys.argv[1],
+                keyword_path=keyword_paths[0],
+                context_path=context_path(meta.context, meta.language),
+                porcupine_model_path=pv_porcupine_model_path_by_language(meta.language),
+                rhino_model_path=pv_rhino_model_path_by_language(meta.language),
+                wake_word_callback=meta.wake_word_callback,
+                inference_callback=meta.inference_callback)])
 
     @classmethod
     def tearDownClass(cls):
-        cls._pv.delete()
+        for meta, pv in cls._pvs:
+            pv.delete()
 
-    def test_process(self):
-        PicovoiceTestCase._is_wake_word_detected = False
-        PicovoiceTestCase._inference = None
+    def run_picovoice(self, pv_index, audio_file_name, intent, slots):
+        _meta, _pv = self._pvs[pv_index]
+        _meta.reset()
 
         audio, sample_rate = \
             soundfile.read(
-                os.path.join(os.path.dirname(__file__), '../../resources/audio_samples/picovoice-coffee.wav'),
+                os.path.join(os.path.dirname(__file__), '../../resources/audio_samples', audio_file_name),
                 dtype='int16')
 
-        for i in range(len(audio) // PicovoiceTestCase._pv.frame_length):
-            frame = audio[i * PicovoiceTestCase._pv.frame_length:(i + 1) * PicovoiceTestCase._pv.frame_length]
-            PicovoiceTestCase._pv.process(frame)
-
-        self.assertTrue(PicovoiceTestCase._is_wake_word_detected)
-        self.assertEqual(PicovoiceTestCase._inference.intent, 'orderBeverage')
-        self.assertEqual(PicovoiceTestCase._inference.slots, dict(size='large', beverage='coffee'))
-
-    def test_process_again(self):
-        self.test_process()
+        for i in range(len(audio) // _pv.frame_length):
+            frame = audio[i * _pv.frame_length:(i + 1) * _pv.frame_length]
+            _pv.process(frame)
+
+        self.assertTrue(_meta.is_wake_word_detected)
+        self.assertEqual(_meta.inference.intent, intent)
+        self.assertEqual(_meta.inference.slots, slots)
+
+    def test(self):
+        self.run_picovoice(
+                pv_index=0,
+                audio_file_name='picovoice-coffee.wav',
+                intent='orderBeverage',
+                slots=dict(size='large', beverage='coffee')) 
+
+    def test_again(self):
+        self.test()                        
+
+    def test_es(self):
+        self.run_picovoice(
+                pv_index=1,
+                audio_file_name='manzana-luz_es.wav',
+                intent='changeColor',
+                slots=dict(location='habitación', color='rosado'))
+
+    def test_de(self):
+        self.run_picovoice(
+                pv_index=2,
+                audio_file_name='heuschrecke-beleuchtung_de.wav',
+                intent='changeState',
+                slots=dict(state='aus'))
+
+    def test_es_again(self):
+        self.test_es()                
+
+    def test_de_again(self):
```

Comment:
Do we not test french because of the Windows UTF-8 file name issue? If so, be sure to add that to tracking, so that we add it once we have released the fix for that.

Reply:
yes exactly, I've added it in the tracking.